src/plugins/augeas/augeas.c

```diff
@@ -216,9 +216,6 @@ static int foreachAugeasNode (augeas * handle, const char * treePath, ForeachAug
 
 		result = (*callback) (handle, curPath, callbackData);
 
-		/* handle the subtree */
-		result = foreachAugeasNode (handle, curPath, callback, callbackData);
```

Comment:
At least the tests say so :). This can be done because ```path//*``` is used instead of ```path/*````. While ```path/*``` returns only the nodes below path, ```path//*``` returns all subtrees below path. See https://github.com/hercules-team/augeas/wiki/Path-expressions for more details.